.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,76 @@
+name: Benchmarks
+
+on:
+  pull_request:
+
+jobs:
+  benchmarks:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Who triggered this?
+        run: |
+          echo "Action triggered by ${{ github.event.pull_request.html_url }}"
+
+      - name: Setup LLVM (Linux)
+        run: |
+          wget https://apt.llvm.org/llvm.sh
+          chmod +x ./llvm.sh
+          sudo ./llvm.sh 11
+          sudo apt install -y llvm-11-dev libclang-11-dev liblldb-11-dev
+          echo "LLVM_INSTALL_PATH=/usr/lib/llvm-11" >> $GITHUB_ENV
+          echo "CC=gcc-10" >> $GITHUB_ENV
+          echo "CXX=g++-10" >> $GITHUB_ENV
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 50 # this is to make sure we obtain the target base commit
+
+      - name: Setup benchmarks
+        run: |
+          echo "BASE_SHA=$(echo ${{ github.event.pull_request.base.sha }} | cut -c1-8)" >> $GITHUB_ENV
+          echo "HEAD_SHA=$(echo ${{ github.event.pull_request.head.sha }} | cut -c1-8)" >> $GITHUB_ENV
+
+          echo "BASELINE_JSON=$(mktemp)" >> $GITHUB_ENV
+          echo "CONTENDER_JSON=$(mktemp)" >> $GITHUB_ENV
+          echo "PR_COMMENT=$(mktemp)" >>  $GITHUB_ENV
+
+      - name: Run benchmarks
+        run: |
+          git checkout ${{ github.event.pull_request.base.sha }}
+          bazel run -c opt lldb-eval:eval_benchmark -- --benchmark_format=json > ${{ env.BASELINE_JSON }}
+
+          git checkout ${{ github.event.pull_request.head.sha }}
+          bazel run -c opt lldb-eval:eval_benchmark -- --benchmark_format=json > ${{ env.CONTENDER_JSON }}
+
+      - name: Compare results
+        run: |
+          git clone --depth 1 --branch "v1.5.3" https://github.com/google/benchmark google_benchmark
+          cd google_benchmark
+
+          # Use python3 to run the compare script, because numpy already dropped
+          # support for python2. The compare script fails with the current
+          # version of numpy because of the non-ASCII symbol in the source code:
+          # https://github.com/numpy/numpy/blob/6d7b8aaed5ae9f0435764675ebac8c9ada06738f/numpy/__init__.py#L292
+          sed -i 's/PY2/PY3/g' tools/BUILD.bazel
+
+          bazel run //tools:compare -- --no-color benchmarks ${{ env.BASELINE_JSON }} ${{ env.CONTENDER_JSON }} | tee cmp_results
+
+          echo 'Benchmark comparison for [`${{ env.BASE_SHA }}`](${{ github.event.repository.html_url }}/commit/${{ github.event.pull_request.base.sha }}) (base) vs [`${{ env.HEAD_SHA }}`](${{ github.event.repository.html_url }}/commit/${{ github.event.pull_request.head.sha }}) (PR)' >> pr_comment
```

Comment:
It would be cool if this was formatted as a markdown table, but let's not complicate it now :)

Reply:
Yeah, that would be nice :) The compare script can actually output the results as JSON (via `--dump_to_json` flag), so it's possible to format the results in a nicer way. It would be also cool to colorize the output (e.g. green for improvements and red for degradations) and add some warning for significant changes.